src/core/server/config/raw_config_service.ts

```diff
@@ -44,34 +44,31 @@ export class RawConfigService {
 
   private readonly config$: Observable<RawConfig>;
 
-  constructor(readonly configFile: string) {
+  constructor(
+    readonly configFiles: ReadonlyArray<string>,
+    configAdapter: (rawValue: Record<string, any>) => RawConfig = rawValue =>
```

Comment:
__note:__ I need this "middleware" to be able to [apply overrides to config](https://github.com/elastic/kibana/pull/22190/files#diff-38fc132b569ae49aebdb69d5421576afR207) before it will be published to any subscriber.